engine/runtime/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -145,6 +175,30 @@ public FunctionCallInfo(FunctionCallInstrumentationNode.FunctionCall call) {
       }
     }
 
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
```

Comment:
Given that you're casting on the next line, maybe this would be better expressed as an explicit `instanceof` check?

Reply:
I did a quick search, and people tend to prefer `getClass` to the `instanceof`. While ones say that is less error-prone when it comes to subclassing, others say that comparing classes can be slightly faster performance-wise. 
This implementation was generated by IntelliJ and I believe it should be battle-tested at this point